### PR TITLE
fix: make error more user friendly

### PR DESCRIPTION
**Pull request by StarlightIbuki**
FTI-4822

### CODE REVIEW

File: lib/resty/mlcache.lua [hunk 1]
```diff
@@ -716,7 +722,9 @@ local function run_callback(self, key, shm_key, data, ttl, neg_ttl,
     -- finished to run the callback
 
     if lerr == "timeout" then
-        local errmsg = "could not acquire callback lock: timeout"
+        local errmsg = "refreshing cache failed: a previous trial is still" ..
+                       " running and we timed out waiting for it to finish." ..
+                       " (key: " .. key .. ")"
```
Comment (thibaultcha) [inline lib/resty/mlcache.lua:727]: I do not like this change. Let's keep the original error and add the `key: ...` suffix, that would be enough.

File: lib/resty/mlcache.lua [hunk 2]
```diff
@@ -665,10 +665,15 @@ local function check_opts(self, opts)
 end
 
 
-local function unlock_and_ret(lock, res, err, hit_lvl)
+local function unlock_and_ret(lock, res, err, hit_lvl, key)
```
Comment (thibaultcha) [inline lib/resty/mlcache.lua:668]: This new argument might not be necessary. lua-resty-lock objects have `self.key` once `:lock()` is called: https://github.com/openresty/lua-resty-lock/blob/master/lib/resty/lock.lua#L161 - let's try to leverage it instead.

**Review by thibaultcha — COMMENTED** [review]
Are there no tests in the suite that catch those errors? If there are, let's include this new suffix. I have not checked.

File: lib/resty/mlcache.lua [hunk 3]
```diff
@@ -697,7 +703,7 @@ local function run_callback(self, key, shm_key, data, ttl, neg_ttl,
                                                                 shm_key,
                                                                 l1_serializer)
         if err then
-            return unlock_and_ret(lock, nil, err)
+            return unlock_and_ret(lock, nil, err, nil)
```
Comment (thibaultcha) [inline lib/resty/mlcache.lua:706]: All these changes are completely unnecessary.

Comment (StarlightIbuki) [inline lib/resty/mlcache.lua:706]:
```suggestion
            return unlock_and_ret(lock, nil, err)
```

File: lib/resty/mlcache.lua [hunk 4]
```diff
@@ -741,7 +748,7 @@ local function run_callback(self, key, shm_key, data, ttl, neg_ttl,
     local pok, perr, err, new_ttl = xpcall(cb, traceback, ...)
     if not pok then
         return unlock_and_ret(lock, nil, "callback threw an error: " ..
-                              tostring(perr))
+                              tostring(perr), nil)
```
Comment (StarlightIbuki) [inline lib/resty/mlcache.lua:751]:
```suggestion
                              tostring(perr))
```

File: lib/resty/mlcache.lua [hunk 5]
```diff
@@ -752,7 +759,7 @@ local function run_callback(self, key, shm_key, data, ttl, neg_ttl,
 
         -- no stale data nor desire to resurrect it
         if not went_stale or not resurrect_ttl then
-            return unlock_and_ret(lock, perr, err)
+            return unlock_and_ret(lock, perr, err, nil)
```
Comment (StarlightIbuki) [inline lib/resty/mlcache.lua:762]:
```suggestion
            return unlock_and_ret(lock, perr, err)
```

File: lib/resty/mlcache.lua [hunk 6]
```diff
@@ -803,7 +810,7 @@ local function run_callback(self, key, shm_key, data, ttl, neg_ttl,
     data, err = set_shm_set_lru(self, key, shm_key, data, ttl, neg_ttl, nil,
                                 shm_set_tries, l1_serializer)
     if err then
-        return unlock_and_ret(lock, nil, err)
+        return unlock_and_ret(lock, nil, err, nil)
```
Comment (StarlightIbuki) [inline lib/resty/mlcache.lua:813]:
```suggestion
        return unlock_and_ret(lock, nil, err)
```

**Comment by thibaultcha** [comment]
Please also contribute test updates, or this will never be merged.

**Comment by StarlightIbuki** [comment]
> Please also contribute test updates, or this will never be merged.

Updated. Please approve the workflow.
P.S. Is the key naming expected? I see no separation between the mlcache object name and the key name.

**Pull request closed without merging**